### ISSUE TITLE
CameraMLVision close event stream exception

### ISSUE THREAD

**Issue by lewixlabs** [santetis/flutter_camera_ml_vision#9]
Hi,
probably I'm doing something wrong.
CameraMLVision is set dinamically by a generic widget, calling setState

```
Widget _activeWidget = CameraMlVision<VisionText>(
        detector: FirebaseVision.instance.textRecognizer().processImage,
        onResult: (VisionText visionTextResult) {
          //print("******* visionText => ${visionTextResult.text}");
          if (!_isScanning && visionTextResult.text.length > 0) {
            var priceDetected = _regexCapturePrice(visionTextResult.text);
            if (priceDetected != null) {
              setState(() {
                _activeWidget = _priceResultView(priceDetected);
              });
            }
          }
        });
```
As you can see in this code, when result are returned, _activeWidget becomes a report widget with details.

Doing this, I get this exception:

> E/EventChannel#plugins.flutter.io/camera/imageStream(26834): Failed to close event stream
> E/EventChannel#plugins.flutter.io/camera/imageStream(26834): java.lang.NullPointerException: Attempt to invoke virtual method 'void android.media.ImageReader.setOnImageAvailableListener(android.media.ImageReader$OnImageAvailableListener, android.os.Handler)' on a null object reference
> E/EventChannel#plugins.flutter.io/camera/imageStream(26834): 	at io.flutter.plugins.camera.CameraHandler$Camera$9.onCancel(CameraHandler.java:827)
> E/EventChannel#plugins.flutter.io/camera/imageStream(26834): 	at io.flutter.plugin.common.EventChannel$IncomingStreamRequestHandler.onCancel(EventChannel.java:194)
> E/EventChannel#plugins.flutter.io/camera/imageStream(26834): 	at io.flutter.plugin.common.EventChannel$IncomingStreamRequestHandler.onMessage(EventChannel.java:162)
> E/EventChannel#plugins.flutter.io/camera/imageStream(26834): 	at io.flutter.view.FlutterNativeView$PlatformMessageHandlerImpl.handleMessageFromDart(FlutterNativeView.java:188)
> E/EventChannel#plugins.flutter.io/camera/imageStream(26834): 	at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage(FlutterJNI.java:202)
> E/EventChannel#plugins.flutter.io/camera/imageStream(26834): 	at android.os.MessageQueue.nativePollOnce(Native Method)
> E/EventChannel#plugins.flutter.io/camera/imageStream(26834): 	at android.os.MessageQueue.next(MessageQueue.java:323)
> E/EventChannel#plugins.flutter.io/camera/imageStream(26834): 	at android.os.Looper.loop(Looper.java:143)
> E/EventChannel#plugins.flutter.io/camera/imageStream(26834): 	at android.app.ActivityThread.main(ActivityThread.java:7225)
> E/EventChannel#plugins.flutter.io/camera/imageStream(26834): 	at java.lang.reflect.Method.invoke(Native Method)
> E/EventChannel#plugins.flutter.io/camera/imageStream(26834): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
> E/EventChannel#plugins.flutter.io/camera/imageStream(26834): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
> 
I attached
[log.txt](https://github.com/santetis/flutter_camera_ml_vision/files/3083952/log.txt)
 full log, if you need.

Is there a graceful way to close CameraVision widget?
Or valid way is calling 
`Navigator.of(context).pop...`
like in your sample?

Thank you again for your help.
Lewix

**Comment by kevin-sakemaer** [comment]
`Navigator.of(context).pop();` should do everything for you.
If you push a new route the plugin will automatically pause itself

**Comment by jaumard** [comment]
hi @lewixlabs thanks for the report ! Are you able to reproduce this with the example app ? Would be nice if you can provide us sample to reproduce it if you don't. Because here you error is native, and we didn't do any native parts on this plugin ^^ so my guess here is that the camera plugin is crashing, but no idea why

**Comment by lewixlabs** [comment]
@Kleak @jaumard thanks for your fast replies :)
As first thing I will change my pattern to use
`Navigator.of(context).pop()`
as suggested.
Changing dinamically a cameravision widget maybe is an unhandled way and native camera is not closed correctly.
After that, if I will encounter again this issue, I will report it.

Thanks!

**Comment by jaumard** [comment]
Any news @lewixlabs ? :)

**Comment by lewixlabs** [comment]
Hi @jaumard ,
yes I was testing in different ways.

To avoid any doubt, I tested your official example in this repository, using also Android Studio (I usually 
work with VSCode) but exception
`setOnImageAvailableListener`
happens again.
Only one difference I applied, is changing package name in android app to make it valid for my google  mlkit json config file.

I attached Android Studio screenshot
![image](https://user-images.githubusercontent.com/7371215/56278846-4ada5a00-6107-11e9-935d-3b77ce36b162.png)
The issue seems not blocking, app doesn't crash & I can scan further barcodes.
The issue happened on my Galaxy S8 with Android 9 & Galaxy Note 3.
If you need more infos let me know.

Lewix

**Comment by kevin-sakemaer** [comment]
fixed in 2.0.1
